developer_architecture.rst

```diff
@@ -86,18 +86,18 @@ steps are built using their corresponding methods in the
 There are five types of :ref:`ExecutionSteps <execution_step>` used to build
 a CloudSlang :ref:`ExecutionPlan <execution_plan>`:
 
--  Start Step
+-  Start
+-  End 
+-  Begin Step
 -  End Step
--  Begin Task Step
--  End Task Step
--  Action Step
+-  Action
 
 An operation's :ref:`ExecutionPlan <execution_plan>`
 is built from a Start Step, an Action Step and an End Step.
 
 A flow's :ref:`ExecutionPlan <execution_plan>` is
-built from a Start Step, a series of Begin Task Steps and End Task
-Steps, and an End Step. The task steps hand off the execution to other
+built from a Start Step, a series of Begin Step Steps and End Step
+Steps, and an End Step. The step steps hand off the execution to other
```

Comment:
yepp, the step of a flow sounds very similar to the execution step of an execution plan -- maybe you should use something like `begin step execution step`